Research: Radio Button (#349)
openui/open-ui: research/src/pages/radio-button.research.mdx

---
menu: Analysis
name: Radio Button
path: /components/radio-button.research
---

import Image from '../components/image'
import Anatomy from '../components/anatomy'
import Concepts from '../components/concepts'
import ComponentCoverage from '../components/component-coverage'

<ComponentCoverage component="Radio Button" />

## Introduction

> _(For a lenghtier discussion, on which this page was based, please refer to the [original radio button issue](https://github.com/WICG/open-ui/issues/251))_

Reviewer: ```suggestion
_(For a lengthier discussion, on which this page was based, please refer to the [original radio button issue](https://github.com/WICG/open-ui/issues/251))_
```

> The research showed that different DS’s treat radio buttons in one of two ways, i.e.:

Reviewer: ```suggestion
The research showed that different design systems treat radio buttons in one of two ways, i.e.:
```

> - as the main component, hosted on a “**radio button group**” component;

Reviewer: I don't understand the difference between these bullets, maybe this can be reworded to be a bit more clear?

Author: I reworded it, maybe it's easier to understand now.

- as a part of a “**radio button group**” component, which, in fact, is the actual component.

> The two approaches seem very similar at first, but the former seems to indicate that single “radio buttons” could exist. The latter, however, shifts the scope up to the “radio button group”, which, although nonexistent as an HTML element, is an important concept for the W3 specification:

Reviewer: ```suggestion
The two approaches seem very similar at first, but the former seems to indicate that single “radio buttons” could exist to indicate a binary state (on or off). The latter, however, shifts the scope up to the “radio button group”, which, although nonexistent as an HTML element, is an important concept for the W3 specification:
```

Author: Hi @una !
You see how this is kind of pointless, because binary states can be accomplished by using checkboxes. I tend to think people do this by mistake, so I would reword your suggestion to "(to **probably** indicate a binary state (on off))".
What do you think?
